currency/forexprovider/exchangerate.host/exchangerate.go

```diff
@@ -0,0 +1,262 @@
+package exchangeratehost
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/currency/forexprovider/base"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+)
+
+// A client for the exchangerate.host API. NOTE: The format and callback
+// parameters aren't supported as they're not needed for this implementation.
+// Furthermore, the source is set to "ECB" as default
+
+const (
+	timeLayout          = "2006-01-02"
+	exchangeRateHostURL = "https://api.exchangerate.host"
+)
+
+var (
+	// DefaultSource uses the ecb for forex rates
+	DefaultSource = "ecb"
+)
+
+// Setup sets up the ExchangeRateHost config
+func (e *ExchangeRateHost) Setup(config base.Settings) error {
+	e.Name = config.Name
+	e.Enabled = config.Enabled
+	e.RESTPollingDelay = config.RESTPollingDelay
+	e.Verbose = config.Verbose
+	e.PrimaryProvider = config.PrimaryProvider
+	e.Requester = request.New(e.Name,
+		common.NewHTTPClientWithTimeout(base.DefaultTimeOut))
+	return nil
+}
+
+// GetLatestRates returns a list of forex rates based on the supplied params
+func (e *ExchangeRateHost) GetLatestRates(baseCurrency, symbols string, amount float64, places int64, source string) (*LatestRates, error) {
+	v := url.Values{}
+	if baseCurrency != "" {
+		v.Set("base", baseCurrency)
+	}
+
+	if symbols != "" {
+		v.Set("symbols", symbols)
+	}
+
+	if amount != 0 {
+		v.Set("amount", strconv.FormatFloat(amount, 'f', -1, 64))
+	}
+
+	if places != 0 {
+		v.Set("places", strconv.FormatInt(places, 10))
+	}
+
+	targetSource := DefaultSource
+	if source != "" {
+		targetSource = source
+	}
+	v.Set("source", targetSource)
+
+	var l LatestRates
+	return &l, e.SendHTTPRequest("latest", v, &l)
+}
+
+// ConvertCurrency converts a currency based on the supplied params
+func (e *ExchangeRateHost) ConvertCurrency(from, to string, date time.Time, baseCurrency, symbols string, amount float64, places int64, source string) (*ConvertCurrency, error) {
```

Comment:
Incredibly ~petty~ minor, but could just make it
`ConvertCurrency(from, to, baseCurrency, symbols, source string, date time.Time, amount float64, places int64)`
to minimise `string`. Can resolve if for variable flow reasons

Reply:
Done, this was just to mirror the API ordering at the time to ensure everything was supported correctly (minus callback and format)